File: Wisehat.Domain/Entities/Widget.cs
// Author: Alexander Dulemba
// Copyright 2023

using System.Numerics;
using System.Text.Json.Serialization;

namespace Wisehat.Domain.Entities;

[JsonConverter(typeof(JsonWidgetConverter))]
public class Widget
{
  public Guid Id { get; set; }

  public string? Description { get; set; }

  public WidgetType Type { get; set; }

  [JsonConverter(typeof(JsonVector2Converter))]
  public Vector2 GrabPosition { get; set; }

  [JsonConverter(typeof(JsonVector2Converter))]
  public Vector2 Size { get; set; }

  public string? Content { get; set; }

  public string BackgroundColor { get; set; } = "#252525";

  public string Border { get; set; } = "1px solid #252525";

  [JsonConverter(typeof(JsonVector2Converter))]
  public Vector2 Position { get; set; }
}

File: Wisehat.Web/wwwroot/js/widget.js
export class Widget {
  constructor(id, description, type, grabPosition, size, content, backgroundColor, borderColor, location) {
    this.id = id;
    this.description = description;
    this.type = type;
    this.grabPosition = grabPosition;
    this.size = size;
    this.content = content;
    this.backgroundColor = backgroundColor;
    this.borderColor = borderColor;
    this.position = location;
  }
}

window.Widget = Widget;